smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -820,6 +944,124 @@ private void readHeaders(
         return payloadBindings;
     }
 
+    private void generateEventStreamDeserializer(GenerationContext context, MemberShape member, Shape target) {
+        TypeScriptWriter writer = context.getWriter();
+        writer.openBlock("context.eventStreamMarshaller.deserialize(", ");", () -> {
+            writer.write("output.body,");
+            writer.openBlock("async event => {", "}", () -> {
+                writer.write("const eventName = Object.keys(event)[0];");
+                writer.write("const eventValue = event[eventName];");
+                writer.openBlock("const eventHeader = Object.entries(eventValue.headers).reduce(",
+                        ");", () -> {
+                            writer.write(
+                                    "(accummulator, curr) => "
+                                            + "{accummulator[curr[0]] = curr[1].value; return accummulator; },"
+                            );
+                            writer.write("{} as {[key: string]: any}");
+                        });
+                writer.openBlock("const eventMessage = {", "};", () -> {
+                    writer.write("headers: eventHeader,");
+                    writer.write("body: event[eventName].body");
+                });
+                writer.openBlock("const parsedEvent = {", "};", () -> {
+                    writer.write("[eventName]: eventMessage");
+                });
+                Symbol targetSymbol = context.getSymbolProvider().toSymbol(target);
+                if (target instanceof StructureShape) {
```

Comment:
See above comment about using a `StringBuilder` to simplify this.

Reply:
fixed.